src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -1025,6 +1058,47 @@ private static string GetFirstLineSubString(string stringToComplete, out bool ha
             return keyValuePairWithCursor;
         }
 
+        private List<CompletionResult> GetResultForEnum(
+            Type type,
+            CompletionContext completionContext)
+        {
+            var stringToComplete = string.Empty;
+            if (completionContext.TokenAtCursor != null)
+            {
+                stringToComplete = completionContext.TokenAtCursor.Text;
+            }
+
+            var quote = '\'';
+            if (stringToComplete.StartsWith('"'))
+            {
+                quote = '"';
+            }
+
+            var allValues = new List<string>();
+            foreach (var value in Enum.GetValues(type))
+            {
+                allValues.Add(quote + value.ToString() + quote);
+            }
+
+            allValues.Sort();
+            string matchString = stringToComplete + "*";
+            var wildcardPattern = WildcardPattern.Get(matchString, WildcardOptions.IgnoreCase | WildcardOptions.CultureInvariant);
+
+            var matchedResults = allValues.Where(r => wildcardPattern.IsMatch(r));
+            if (matchedResults == null || !matchedResults.Any())
+            {
+                matchedResults = allValues;
+            }
+
+            var result = new List<CompletionResult>();
+            foreach (var value in matchedResults)
```

Comment:
```suggestion
            foreach (string value in matchedResults)
```

Reply:
`value` and `result` could maybe be more descriptively named

Reply:
result is used throughout this source so leaving that, will rename value